apps/web/src/stores/documentVersions.ts

```diff
@@ -111,8 +119,46 @@ export default function useDocumentVersions(
         }
       }
     },
-    [executeDestroyDocument, mutate, data],
+    [executeDestroyDocument, mutate, data, currentDocument?.documentUuid],
+  )
+
+  const destroyFolder = useCallback(
+    async (path: string) => {
+      const [_, error] = await executeDestroyFolder({
+        projectId: project.id,
+        commitId: commit.id,
+        path,
+      })
+
+      if (error) {
+        toast({
+          title: 'Error deleting folder',
+          description: error.formErrors?.[0] || error.message,
+          variant: 'destructive',
+        })
+      } else {
+        mutate()
```

Comment:
you are not waiting for this mutation before redirecting could lead to jumpy ui

Reply:
Good catch